plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/wizards/NewInterfaceWizardPage.java

```diff
@@ -0,0 +1,173 @@
+package org.eclipse.reddeer.eclipse.jdt.ui.wizards;
+
+import org.eclipse.reddeer.common.exception.RedDeerException;
+import org.eclipse.reddeer.core.exception.CoreLayerException;
+import org.eclipse.reddeer.core.reference.ReferencedComposite;
+import org.eclipse.reddeer.jface.wizard.WizardPage;
+import org.eclipse.reddeer.swt.impl.button.CheckBox;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.table.DefaultTable;
+import org.eclipse.reddeer.swt.impl.text.DefaultText;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Wizard page for creating a java interface.
+ * 
+ * @author zcervink@redhat.com
+ * 
+ */
+public class NewInterfaceWizardPage extends WizardPage {
+
+	/**
+	 * Instantiates a new new java interface wizard page.
+	 */
+	public NewInterfaceWizardPage(ReferencedComposite referencedComposite) {
+		super(referencedComposite);
+	}
+
+	/**
+	 * Sets a given name.
+	 * 
+	 * @param name Name
+	 */
+	public NewInterfaceWizardPage setName(String name) {
+		new LabeledText(this, "Name:").setText(name);
+		return this;
+	}
+
+	/**
+	 * Sets a given package name.
+	 * 
+	 * @param packageName Package name
+	 */
+	public NewInterfaceWizardPage setPackage(String packageName) {
+		new LabeledText(this, "Package:").setText(packageName);
+		return this;
+	}
+
+	/**
+	 * Sets a given source folder.
+	 * 
+	 * @param sourceFolder Source folder
+	 */
+	public NewInterfaceWizardPage setSourceFolder(String sourceFolder) {
+		new LabeledText(this, "Source folder:").setText(sourceFolder);
+		return this;
+	}
+
+	/**
+	 * Returns a package name.
+	 * 
+	 * @return package name
+	 */
+	public String getPackage() {
+		return new LabeledText(this, "Package:").getText();
+	}
+
+	/**
+	 * Returns a interface name.
+	 * 
+	 * @return Interface name
+	 */
+	public String getName() {
+		return new LabeledText(this, "Name:").getText();
+	}
+
+	/**
+	 * Returns a source folder.
+	 * 
+	 * @return Source folder
+	 */
+	public String getSourceFolder() {
+		return new LabeledText(this, "Source folder:").getText();
+	}
+
+	/**
+	 * Check/Uncheck Enclosing type.
+	 *
+	 * @param toggle enclosing type checkbox
+	 */
+	public NewInterfaceWizardPage toggleEnclosingTypeCheckBox(boolean toggle) {
+		new CheckBox("Enclosing type:").toggle(toggle);
+		return this;
+	}
+
+	/**
+	 * Returns Enclosing type.
+	 * 
+	 * @return Enclosing type
+	 */
+	public String getEnclosingType() {
+		return new DefaultText(2).getText();
+	}
+
+	/**
+	 * Sets a given Enclosing type.
+	 * 
+	 * @param enclosingTypeName Enclosing type
+	 */
+	public NewInterfaceWizardPage setEnclosingType(String enclosingTypeName) {
+		new DefaultText(2).setText(enclosingTypeName);
+		return this;
+	}
+
+	/**
+	 * Add extended interface
+	 * 
+	 * @param interfaceName String with name of interface to add
+	 */
+	public void addExtendedInterface(String interfaceName) {
+		new PushButton(3).click();
```

Comment:
After this should be check, if correct window opens, like new DefaultShell(...).

Reply:
added.